schematics/src/helpers/override/factory.ts

```diff
@@ -19,9 +20,15 @@ export function override(options: Options): Rule {
     const workspace = await getWorkspace(host);
     const project = workspace.projects.get(options.project);
     const sourceRoot = project.sourceRoot;
-    const from = `${
-      options.path ? `${options.path}/` : !options.from?.startsWith(`${sourceRoot}/app/`) ? `${sourceRoot}/app/` : ''
-    }${options.from.replace(/\/$/, '')}`;
+    const from = normalize(
+      `${
+        options.path
```

Comment:
I would not use path sep and stick to slashes only.
just do `normalize(options.path).replace(/\\/, '/')` and keep the remaining.